### PR TITLE
add copy method to drive.py

### LINKED ISSUE #83: when to implement files.copy method?
Copy method is  very useful. Every time I need it , I have to do like this:
        body = {
            "parents": [{"kind": "drive#fileLink",
                         "id": target_folder_id}],
            'title': newtitle
            }
          drive.auth.service.files().copy(fileId=source_file_id, supportsAllDrives=True,body=body).execute() 

but I just want to like this:
    drive.copy(souce_file_id,target_folder_id)

### PR DESCRIPTION
**Pull request by simone-viozzi**
Hi i added the copy method keeping it as simple as possible.

this PR fixes #83 #53

### CODE REVIEW

File: pydrive2/drive.py [hunk 1]
```diff
@@ -47,3 +47,29 @@ def GetAbout(self):
...
+        :param source_file_id: ID of the file to copy.
+        :type source_file_id: str.
+        :param target_folder_id: ID of the folder to copy to.
+        :type target_folder_id: str.
+        :param new_name: Name of the new file.
+        :type new_name: str.
+        """
+
+        body = {
+            "parents": [{"kind": "drive#fileLink",
```
Comment (shcheklein) [inline pydrive2/drive.py:65]: Q: `drive#fileLink` why do we need specifying `kind` here?

Comment (simone-viozzi) [reply pydrive2/drive.py:65]: you are right pointing it out, because we don't. It works with just the id.

File: pydrive2/drive.py [hunk 2]
```diff
@@ -47,3 +47,28 @@ def GetAbout(self):
         :returns: A dictionary of Google Drive information like user, usage, quota etc.
         """
         return self.auth.service.about().get().execute(http=self.http)
+        
+
+    @LoadAuth
+    def CopyFile(self, source_file_id, target_folder_id, new_name):
```
Comment (shcheklein) [inline pydrive2/drive.py:53]: I this it should be part of the `files.py` and `GoogleDriveFile` class. 

It won't have `source_file_id` then, only target path (folder) and new name. Not yet sure if we want to accept `target_folder_id` as an id or as another `GoogleDriveFile` instance (I would probably do latter) + regular things like `params`, etc that are part of the interface.

also, when we agree on the interface, let's please add a test for this

**Review by shcheklein — CHANGES_REQUESTED** [review]
Thanks for the PR! I've left a comment. Primarily next step is to move it to the `GoogleDriveFile` class and we'll need to add tests.

File: pydrive2/files.py [hunk 3]
```diff
@@ -518,6 +518,30 @@ def Delete(self, param=None):
         """
         self._FilesDelete(param=param)
 
+    @LoadAuth
+    def Copy(self, target_folder: "GoogleDriveFile", new_title = None):
```
[the remark this answers is not in the document]
Comment (shcheklein) [reply pydrive2/files.py:522]: good question.

The way you would create a new folder is:

```python
item = self.client.CreateFile(
            {"title": title, "parents": [parent], "mimeType": FOLDER_MIME_TYPE}
        )
item.Upload()
```

where `FOLDER_MIME_TYPE = "application/vnd.google-apps.folder"`

so, `mimeType` is required in that case.

Is it required by the GDrive API in this case? If yes, we should pass it and check that `target_folder` has it also. 

What happens if we pass a file ID to the `files().copy()`?

File: pydrive2/files.py [hunk 4]
```diff
@@ -518,6 +518,30 @@ def Delete(self, param=None):
...
+
+        if new_title is None:
+            new_title = self["title"]
+
+        body = {
+            "parents": [{"id": target_folder['id']}],
+            'title': new_title 
+            }
+            
+        self.auth.service.files().copy(
```
Comment (shcheklein) [inline pydrive2/files.py:539]: PTAL at the other methods around, this call should be wrapped with `try ... except errors.HttpError as error: ... raise ApiRequestError(error)`

File: pydrive2/files.py (hunk 3 above)
Comment (shcheklein) [inline pydrive2/files.py:522]: should be an extra argument: `param = None`

let's adhere to the existing code and not do `: "GoogleDriveFile"`

see if the new_title can be passed as part of the `params`

**Review by shcheklein — CHANGES_REQUESTED** [review]
good progress :) please, see some comments + we still need to add tests for this

File: pydrive2/files.py (hunk 3 above)
Comment (simone-viozzi) [reply pydrive2/files.py:522]: If i understood the use of `param` correctly we should have:

```python
self.auth.service.files().copy(**param).execute()
``` 

in this case `new_title` is a part of `body` that is a part of `param`, so the user would have to pass a double dict like:

```python
param = {
    "body": {
        "title": new_title
    }
}
```

and than we fill the other fields inside the method.

This will also leave the possibility for the user to pass any other parameters. 
What do you think? if is just for the `new_title` i feel like is a little over engineered

File: pydrive2/files.py (hunk 4 above)
Comment (simone-viozzi) [reply pydrive2/files.py]: resolved in [748ee9b](https://github.com/iterative/PyDrive2/pull/188/commits/748ee9bb4d4b9178c8876b0f3ffd897c92e2c4b1)

**Comment by simone-viozzi** [comment]
Where the test should go? on `test_file.py` of `test_fs.py`? 
they have very different structure

also, I can't seem to run the existing ones :) but is most likely because of the authentication.

File: pydrive2/drive.py [hunk 5]
```diff
@@ -47,3 +47,4 @@ def GetAbout(self):
         :returns: A dictionary of Google Drive information like user, usage, quota etc.
         """
         return self.auth.service.about().get().execute(http=self.http)
+
```
Comment (shcheklein) [inline pydrive2/drive.py:50]: minor: let's not do these unrelated changes please?

File: pydrive2/files.py (hunk 3 above)
Comment (shcheklein) [reply pydrive2/files.py:522]: check the `InsertPermission` please - it can be probably done this way:

`def Copy(self, target_folder, new_title, param=None):`

param argument is still needed, it allows user to pass some extra configs if needed

File: pydrive2/drive.py (hunk 5 above)
Comment (simone-viozzi) [reply pydrive2/drive.py]: solved in [d39458c](https://github.com/iterative/PyDrive2/pull/188/commits/d39458cdce6b55c416fa13d64e0cdeeef4eae0f7)

File: pydrive2/files.py (hunk 3 above)
Comment (simone-viozzi) [reply pydrive2/files.py]: solved in [327e2eb](https://github.com/iterative/PyDrive2/pull/188/commits/327e2eba0f6b4680d05b045f6139ad5f85320fa0)

File: pydrive2/drive.py [hunk 6]
```diff
@@ -46,4 +46,4 @@ def GetAbout(self):
 
         :returns: A dictionary of Google Drive information like user, usage, quota etc.
         """
-        return self.auth.service.about().get().execute(http=self.http)
+        return self.auth.service.about().get().execute(http=self.http)
```
Comment (shcheklein) [inline pydrive2/drive.py:49]: please, let's not change this - it unrelated to the PR (and actually it's better to have a newline at the EOF)

File: pydrive2/files.py [hunk 7]
```diff
@@ -518,6 +518,36 @@ def Delete(self, param=None):
...
+        :param target_folder: Folder where the file will be copied.
+        :type target_folder: GoogleDriveFile.
+        :param new_title: Name of the new file.
+        :type new_title: str.
+        :param param: addition parameters to pass
+        :type param: dict
+        """
+
+        if new_title is None:
+            new_title = self["title"]
```
Comment (shcheklein) [inline pydrive2/files.py:534]: Question: is it required? I mean, do we have to provide the title? 

I'm asking because this `self["title"]` call might require hitting an API to fetch the metadata (which is an expensive operation).

File: pydrive2/files.py [hunk 8]
```diff
@@ -518,6 +518,36 @@ def Delete(self, param=None):
         """
         self._FilesDelete(param=param)
 
+    @LoadAuth
+    def Copy(self, target_folder, new_title = None, param=None):
```
Comment (shcheklein) [inline pydrive2/files.py:522]: style: `new_title=None`

File: pydrive2/files.py [hunk 9]
```diff
@@ -518,6 +518,36 @@ def Delete(self, param=None):
...
+
+        param["fileId"] = self["id"]
+        param["supportsAllDrives"] = True
+        param["body"] = {}
+
+        param["body"]["parents"] = [{"id": target_folder["id"]}]
+        param["body"]["title"] = new_title
+
+        try:
+            self.auth.service.files().copy(**param).execute()
```
Comment (shcheklein) [inline pydrive2/files.py:547]: it looks like you need to pass `http=self.http` into `execute()`

Comment (shcheklein) [inline pydrive2/files.py:547]: btw, does this call return anything? new file for example? it would be handy to return it then from the function

File: pydrive2/files.py (hunk 8 above)
Comment (shcheklein) [reply pydrive2/files.py:522]: Question: how do I copy to the root of my personal drive or to the root of a shared drive?

File: pydrive2/test/test_file.py [hunk 10]
```diff
@@ -842,6 +842,39 @@ def test_Parallel_Insert_File_Passed_HTTP(self):
         files = self._parallel_uploader(self.FILE_UPLOAD_COUNT, 10)
         self._parallel_downloader(files, 10)
 
+    # Tests for Copy file.
+    # ====================
+
+    def test_CopyFile(self):
+        drive = GoogleDrive(self.ga)
+        file1 = drive.CreateFile()
+        filename = self.getTempFile("copytestfile")
```
Comment (shcheklein) [inline pydrive2/test/test_file.py:851]: `content="hello world"` should be passed here instead of the next line and the `SetContentString`

File: pydrive2/test/test_file.py [hunk 11]
```diff
@@ -842,6 +842,39 @@ def test_Parallel_Insert_File_Passed_HTTP(self):
...
+        content = "hello world!"
+        file1["title"] = filename
+        file1.SetContentString(content)
+        pydrive_retry(file1.Upload)
+        pydrive_retry(file1.FetchMetadata)
```
Comment (shcheklein) [inline pydrive2/test/test_file.py:856]: it should not be needed

File: pydrive2/test/test_file.py [hunk 12]
```diff
@@ -842,6 +842,39 @@ def test_Parallel_Insert_File_Passed_HTTP(self):
...
+
+        parent_id = file1["parents"][0]["id"]
+        parent_folder = drive.CreateFile({"id": parent_id})
+
+        file1.Copy(parent_folder, new_title="copytestfile_copy")
```
Comment (shcheklein) [inline pydrive2/test/test_file.py:861]: so, we copy to the same folder? can we shortcut the API in this case and make parent optional? what happens if you don't pass the parent?

also, it feels it would be more general if we create a temp dir and place the file there

File: pydrive2/test/test_file.py [hunk 13]
```diff
@@ -842,6 +842,39 @@ def test_Parallel_Insert_File_Passed_HTTP(self):
...
+
+        files = drive.ListFile({"q": f"'{parent_id}' in parents"}).GetList()
+
+        file2 = None
```
Comment (shcheklein) [inline pydrive2/test/test_file.py:865]: this whole construct can be replace with something like:

`self.assertIn("copytestfile_copy", [f["title"] for f in files])`

File: pydrive2/test/test_file.py [hunk 14]
```diff
@@ -842,6 +842,39 @@ def test_Parallel_Insert_File_Passed_HTTP(self):
...
+        for f in files:
+            if f["title"] == "copytestfile_copy":
+                file2 = f
+                break
+        
+        self.assertIsNotNone(file2)
+        
+        pydrive_retry(file2.FetchContent)
+        self.assertEqual(file2.GetContentString(), content)
```
Comment (shcheklein) [inline pydrive2/test/test_file.py:874]: should it be `file1.content, content` or something like this?

**Review by shcheklein — CHANGES_REQUESTED** [review]
Good progress! Please take a look at the latest review.

File: pydrive2/files.py (hunk 7 above)
Comment (simone-viozzi) [reply pydrive2/files.py:534]: if you don't provide a title, the filename of the copied file will be "Copy of [old-filename]".
So, yes, we can remove those two lines, if the user want to use the same title it will have to pass it.

File: pydrive2/files.py (hunk 8 above)
Comment (simone-viozzi) [reply pydrive2/files.py:522]: you can't copy a folder, if a user attempt that they will receive:

```python
[WARNING] googleapiclient.http -> _should_retry_response: Encountered 403 Forbidden with reason "cannotCopyFile"
Traceback (most recent call last):
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/files.py", line 547, in Copy
    self.auth.service.files().copy(**param).execute()
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/googleapiclient/_helpers.py", line 130, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/googleapiclient/http.py", line 938, in execute
    raise HttpError(resp, content, uri=self.uri)
googleapiclient.errors.HttpError: <HttpError 403 when requesting https://www.googleapis.com/drive/v2/files/1KL7v_9MSReLZifHhpv8mJG_NY1q9JAXC/copy?supportsAllDrives=true&alt=json returned "This file cannot be copied by the authenticated user.". Details: "[{'domain': 'global', 'reason': 'cannotCopyFile', 'message': 'This file cannot be copied by the authenticated user.'}]">

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/simone/workspace/drive-transfere-ownership/tests/main.py", line 66, in <module>
    file.Copy(folder)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/auth.py", line 82, in _decorated
    return decoratee(self, *args, **kwargs)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/files.py", line 549, in Copy
    raise ApiRequestError(error)
pydrive2.files.ApiRequestError: <HttpError 403 when requesting https://www.googleapis.com/drive/v2/files/1KL7v_9MSReLZifHhpv8mJG_NY1q9JAXC/copy?supportsAllDrives=true&alt=json returned "This file cannot be copied by the authenticated user.". Details: "[{'domain': 'global', 'reason': 'cannotCopyFile', 'message': 'This file cannot be copied by the authenticated user.'}]">
```

To copy a folder you need a recursive function that copy each file and create the folders as needed.

Should we check if the file they are attempting to copy is a folder and raise an error?

File: pydrive2/test/test_file.py (hunk 12 above)
Comment (simone-viozzi) [reply pydrive2/test/test_file.py:861]: i never tried before to not pass a parent folder, but it works. Drive will place the copy in the same folder as the file.

I will also make `target_folder` optional

File: pydrive2/test/test_file.py (hunk 13 above)
Comment (simone-viozzi) [reply pydrive2/test/test_file.py:865]: i also need `file2` so that i can get it's content, to compare it with `content`. Witch is the string inside `file1`.

i will add some comments to make what I'm doing a little clearer

File: pydrive2/files.py (hunk 8 above)
Comment (simone-viozzi) [reply pydrive2/files.py]: same way as any other folder

```python
folder = drive.CreateFile({'id': 'root'})

file.Copy(folder)  
```

i don't know about shared drives btw, i can't do tests on that so i don't know if things are different.

File: pydrive2/test/test_file.py (hunk 14 above)
Comment (simone-viozzi) [reply pydrive2/test/test_file.py:867]: `file1` if the file i copied (original)
`file2` is the copy

i know the content of `file1` because i had set it to `content`: 
```python
file1.SetContentString(content)
```

Now i need to test if the content of the copy is the same as the original:
```python
self.assertEqual(file2.GetContentString(), content)
```

File: pydrive2/test/test_file.py (hunk 12 above)
Comment (simone-viozzi) [reply pydrive2/test/test_file.py]: also i added a new test in witch i create a temp folder and place the copy there

File: pydrive2/files.py (hunk 8 above)
Comment (simone-viozzi) [reply pydrive2/files.py]: > What I was asking about is - what should I pass to this method to copy a _file_ to the _root_ of the Google Drive? or to a shared Google Drive?

sorry, yesterday i read that the wrong way and the answer was off-topic, so i deleted the comment and did a new one with the real answer (see comment above)

File: pydrive2/files.py [hunk 15]
```diff
@@ -518,6 +518,40 @@ def Delete(self, param=None):
...
+
+        if param is None:
+            param = {}
+
+        param["fileId"] = self["id"]
+        param["supportsAllDrives"] = True
+        param["body"] = {}
+
+        if target_folder:
+            param["body"]["parents"] = [{"id": target_folder["id"]}]
```
Comment (shcheklein) [inline pydrive2/files.py:544]: q: can you pass multiple parents?

File: pydrive2/test/test_file.py (hunk 10 above)
Comment (shcheklein) [reply pydrive2/test/test_file.py:855]: @simone-viozzi please check if we can do this ^^

File: pydrive2/test/test_file.py [hunk 16]
```diff
@@ -842,6 +842,67 @@ def test_Parallel_Insert_File_Passed_HTTP(self):
...
+
+        # create a temp file and set it's content to a known string
+        file1 = drive.CreateFile()
+        filename = self.getTempFile("copytestfile")
+        file1["title"] = filename
+        file1.SetContentString(content)
+        pydrive_retry(file1.Upload)
+
+        # copy the file
+        file2 = file1.Copy(new_title="copytestfile_copy")
```
Comment (shcheklein) [inline pydrive2/test/test_file.py:861]: wrap with `pydrive_retry` please

File: pydrive2/test/test_file.py [hunk 17]
```diff
@@ -842,6 +842,67 @@ def test_Parallel_Insert_File_Passed_HTTP(self):
...
+
+        # create a temp file and set it's content to a known string
+        file1 = drive.CreateFile()
+        filename = self.getTempFile("copytestfile")
+        file1["title"] = filename
+        file1.SetContentString(content)
+        pydrive_retry(file1.Upload)
+
+        # create a temp directory
+        temp_dir = drive.CreateFile({
```
Comment (shcheklein) [inline pydrive2/test/test_file.py:884]: pydrive_retry

File: pydrive2/test/test_file.py [hunk 18]
```diff
@@ -842,6 +842,67 @@ def test_Parallel_Insert_File_Passed_HTTP(self):
...
+            'title': "temp_dir",
+            'mimeType': 'application/vnd.google-apps.folder',
+            'parents': [{'id': file1["parents"][0]["id"]}]
+        })
+        temp_dir.Upload()
```
Comment (shcheklein) [inline pydrive2/test/test_file.py:889]: pydrive_retry

File: pydrive2/test/test_file.py [hunk 19]
```diff
@@ -842,6 +842,67 @@ def test_Parallel_Insert_File_Passed_HTTP(self):
...
+
+        # copy the file into the new folder
+        file2 = file1.Copy(target_folder=temp_dir, new_title="copytestfile_copy")
```
Comment (shcheklein) [inline pydrive2/test/test_file.py:892]: pydrive_retry

File: pydrive2/test/test_file.py [hunk 20]
```diff
@@ -842,6 +842,67 @@ def test_Parallel_Insert_File_Passed_HTTP(self):
...
+
+        self.assertIsNotNone(file2)
+
+        pydrive_retry(file2.FetchContent)
+        # assert that the content of the copied file is the same as the original (file1)
+        self.assertEqual(file2.GetContentString(), content)
+
+        files = drive.ListFile({"q": f"'{temp_dir['id']}' in parents"}).GetList()
```
Comment (shcheklein) [inline pydrive2/test/test_file.py:900]: pydrive_retry

File: pydrive2/files.py [hunk 21]
```diff
@@ -518,6 +518,40 @@ def Delete(self, param=None):
         """
         self._FilesDelete(param=param)
 
+    @LoadAuth
+    def Copy(self, target_folder=None, new_title=None, param=None):
+        """Creates a copy of this file. Folders cannot be copied.
+
+        :param target_folder: Folder where the file will be copied.
+        :type target_folder: GoogleDriveFile
+        :param new_title: Name of the new file., defaults to None
```
Comment (shcheklein) [inline pydrive2/files.py:527]: `.,` -> `,`

Comment (shcheklein) [reply pydrive2/files.py:527]: I think we can specify that if parent is also None, `Copy_` prefix is added

Comment (shcheklein) [reply pydrive2/files.py:527]: Actually, we should probably remove this "defaults to None"

**Comment by shcheklein** [comment]
@simone-viozzi please take a look^, especially the pre-commit hook, it means you need to run it locally. Tests - I'll check if I can run them from your branch.

File: pydrive2/test/test_file.py (hunk 10 above)
Comment (simone-viozzi) [reply pydrive2/test/test_file.py:855]: with 
```python
filename = self.getTempFile("copytestfile", content)
```

the test fails because `file1` has no content

File: pydrive2/files.py (hunk 15 above)
Comment (simone-viozzi) [reply pydrive2/files.py:544]: apparently no:

```python
[WARNING] googleapiclient.http -> _should_retry_response: Encountered 403 Forbidden with reason "cannotAddParent"
Traceback (most recent call last):
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/files.py", line 552, in Copy
    self.auth.service.files().copy(**param).execute(http=self.http)
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/googleapiclient/_helpers.py", line 130, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/home/simone/workspace/drive-transfere-ownership/.direnv/python-3.8.12/lib/python3.8/site-packages/googleapiclient/http.py", line 938, in execute
    raise HttpError(resp, content, uri=self.uri)
googleapiclient.errors.HttpError: <HttpError 403 when requesting https://www.googleapis.com/drive/v2/files/1WO_CTrDj62v3RWJ5OP5vCpsNlCKF38mIPQ-nAcog8H0/copy?supportsAllDrives=true&alt=json returned "Increasing the number of parents is not allowed". Details: "[{'domain': 'global', 'reason': 'cannotAddParent', 'message': 'Increasing the number of parents is not allowed'}]">

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/simone/workspace/drive-transfere-ownership/tests/main.py", line 71, in <module>
    f = file.Copy(folders)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/auth.py", line 82, in _decorated
    return decoratee(self, *args, **kwargs)
  File "/home/simone/workspace/drive-transfere-ownership/PyDrive2/pydrive2/files.py", line 555, in Copy
    raise ApiRequestError(error)
pydrive2.files.ApiRequestError: <HttpError 403 when requesting https://www.googleapis.com/drive/v2/files/1WO_CTrDj62v3RWJ5OP5vCpsNlCKF38mIPQ-nAcog8H0/copy?supportsAllDrives=true&alt=json returned "Increasing the number of parents is not allowed". Details: "[{'domain': 'global', 'reason': 'cannotAddParent', 'message': 'Increasing the number of parents is not allowed'}]">
```

maybe this is allowed in some other types of Drive, otherwise it doesn't make sent that `parents` is an array if it can have only one element.

Comment (shcheklein) [reply pydrive2/files.py:544]: thanks!

File: pydrive2/test/test_file.py (hunk 10 above)
Comment (shcheklein) [reply pydrive2/test/test_file.py:855]: it should be `filename = self.getTempFile("copytestfile", content=content)` I think

if it doesn't work - it should be fixed I think

File: pydrive2/files.py [hunk 22]
```diff
@@ -518,6 +518,42 @@ def Delete(self, param=None):
         """
         self._FilesDelete(param=param)
 
+    @LoadAuth
+    def Copy(self, target_folder=None, new_title=None, param=None):
+        """Creates a copy of this file. Folders cannot be copied.
+
+        :param target_folder: Folder where the file will be copied.
+        :type target_folder: GoogleDriveFile
```
Comment (shcheklein) [inline pydrive2/files.py:526]: consistency, if keep `,optional` below, let's do the same here please, if not - let's remove everywhere.

File: pydrive2/test/test_file.py (hunk 10 above)
Comment (simone-viozzi) [reply pydrive2/test/test_file.py:855]: I copied the usage on other tests:

https://github.com/iterative/PyDrive2/blob/2e43e4561d965ce78dc158a02fbdb75ba6c38105/pydrive2/test/test_file.py#L84-L89

but now i searched more and found a usage with content:

https://github.com/iterative/PyDrive2/blob/2e43e4561d965ce78dc158a02fbdb75ba6c38105/pydrive2/test/test_file.py#L222-L230

so to use content on getTempFile i need to:

```python
content = "hello world!"

# create a temp file and set it's content to a known string
file1 = drive.CreateFile()
filename = self.getTempFile("copytestfile", content=content)
file1["title"] = filename
file1.SetContentFile(filename)
pydrive_retry(file1.Upload)
```

And with `SetContentFile` the test runs.

i will update the code than.

**Comment by shcheklein** [comment]
Looks good to me @simone-viozzi , let me try to run tests for this ... and we can merge it soon. Thanks for the PR.

**Comment by shcheklein** [comment]
@simone-viozzi I've rebased the branch, everything runs fine! 

The last thing to see is we need to update docs https://github.com/iterative/PyDrive2/tree/master/docs - could you please take a look?

**Comment by simone-viozzi** [comment]
i think we can add a short example in https://docs.iterative.ai/PyDrive2/filemanagement/

The documentation of the function itself are generated from the doc-string, right?

**Comment by shcheklein** [comment]
> The documentation of the function itself are generated from the doc-string, right?

Yes, but I would see if there are mentions in those `rst`s. If not - that's fine.

If you can do a quick example - that would be great! If not that's also fine- not a blocker. Just let me know when you feel you are ready with that, and we'll merge it.

**Review by shcheklein — APPROVED** [review]

**Comment by shcheklein** [comment]
@simone-viozzi done, thanks for the PR.
